### PR TITLE
Markdown parser

### PR DESCRIPTION
**Pull request by valiafetisov**
Implements custom md parser using [remark](https://github.com/remarkjs/remark).

Example usage:

```ts
// to get list of checks one need to perform
const { tree, currentCheckboxes, defaultCheckboxes } = await parseMd('- [ ] Content of the MD document');

// `defaultCheckboxes` would contain an array of checks one need to perform in this markdown (in their default state)
console.log(defaultCheckboxes)
[
  {
    type: 'listItem',
    spread: false,
    checked: false,
    children: [ [Object] ],
    position: { start: [Object], end: [Object] },
    path: '/children/1/children/0',
    conclusion: undefined
  },
  // ...
]
```
To flip the checkbox, you only need to set `conclusion` to one of the following strings:
- `"correct"` – to indicate that the check is correct ✅
- `"critical"` – to indicate that the check is incorrect ❌
- `"acceptable"` – to indicate that the risk is acceptable ⚠️
- `"not applicable"` – to indicate that this check is ~~not applicable~~

For example: 
```ts
// Set the very first check as correct
defaultCheckboxes[0].conclusion = "correct"

// Render tree back into markdown
const markdown = await renderMd({ tree, checkboxes: defaultCheckboxes });

console.log(markdown)
// '- ✅ Content of the MD document'
```

### CODE REVIEW

**Review by DaeunYoon — APPROVED** [review]
test examples makes sense to me :)

File: packages/nextjs/utils/md-parser.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  const parser = PARSERS.find(c => c.type === node.conclusion);
+  if (parser) {
+    // @ts-expect-error
+    node.children[0].children[0].value = parser.apply(text, parser.marker, node);
+  }
+};
+
+export const parseMd = async function (text: string) {
+  const tree = await unified().use(remarkParse).use(remarkGfm).parse(text);
+  // console.log(JSON.stringify(tree, null, 2));
```
Comment (DaeunYoon) [inline packages/nextjs/utils/md-parser.ts:107]: Just saw the `console.log`. This can be deleted if you feel like :D

Comment (valiafetisov) [reply packages/nextjs/utils/md-parser.ts:107]: I don't think the commented out log have to be deleted in this project. There are also a bunch of completed TODO comments, which I'm fine with 🙂

**Comment by valiafetisov** [comment]
I will continue the work in another PR, if needed (extracting extra comments and potentially rendering html?)

**Pull request merged**